### ISSUE TITLE
Add support for ITextRangeProvider

### ISSUE THREAD

**Issue by chiaramooney** [microsoft/react-native-windows#13350]
Add support for ITextRangeProvider

## Documentation
https://learn.microsoft.com/en-us/dotnet/api/system.windows.automation.provider.itextrangeprovider?view=windowsdesktop-8.0

## Behavior
Exposes methods and properties to support UI Automation client access to a span of continuous text in a text container that implements [ITextProvider](https://learn.microsoft.com/en-us/dotnet/api/system.windows.automation.provider.itextprovider?view=windowsdesktop-8.0).

## Implementation Plan
We should:
1. Implement the ITextRangeProvider APIs in CompositionDynamicAutomationProvider.
1. Set the list of controls who implement the ITextRangeProvider to be Document and Edit/Text under certain conditions. To get more information on the conditions visit the page for the individual UIA control types.

For additional reference, see other provider implementations on Fabric: https://github.com/microsoft/react-native-windows/pull/11874

## To Be Clarified
Unclear if documentation if implementation of ITextRangeProvider should be necessary in all cases of an implemented ITextProvider.
